src/helpers/assertStringsWithinText.ts

```diff
@@ -0,0 +1,30 @@
+import React, { ReactElement } from 'react';
+import { ReactTestInstance } from 'react-test-renderer';
+import { filterNodeByType } from './filterNodeByType';
+
+export const assertStringsWithinText = (instance: ReactTestInstance): void => {
+  const nodesWithStringChild = instance.findAll(hasNodeStringChild);
+  nodesWithStringChild.forEach((node) => {
+    const isHostTextComponent = filterNodeByType(node, 'Text');
+    const isNativeComponent = typeof node.type === 'string';
```

Comment:
This second check seems no longer needed since `finderNodeByType` verifies that `node.type` is string equal to `Text`.
```suggestion
```

Reply:
We're checking !isHostTextComponent & isNativeComponent so it's not equivalent to !isHostComponent, the condition would be false for a component that has a type function or object, whereas with the check isNativeComponent it would be false

Reply:
Can we rename it as `isHostComponent` because seems like `host` and `native` refere to same thing, so it would make sense to use the same pattern as more readable.